File: 1cpp/Parallelogram.cpp
//
// Created by olile on 10.04.24.
//

#include "Parallelogram.h"

#include <cmath>


double Parallelogram::calcArea() {
    return sin(this->angle*M_PI/180)*this->a*this->b;
}

std::string Parallelogram::info(){
    return "Parallelogram";
}

File: 1java/1java/src/Parallelogram.java
// Parallelogram.java (Example implementation)
public class Parallelogram implements Shape {
    private double a, b;
    private double angle;

    public Parallelogram(double a, double b, double angle) {
        this.a = a;
        this.b = b;
        this.angle = angle;
    }

    @Override
    public double calcArea() {
        // Implement the area calculation for a parallelogram
        // Return the actual area value.
        return Math.sin(Math.toRadians(angle)) * a * b;
    }

    @Override
    public double calcCircumference() {
        return 2 * (a + b);
    }

    @Override
    public String info() {
        return "Parallelogram";
    }
}
